src/dmd/dstruct.d

```diff
@@ -69,137 +69,117 @@ extern (C++) FuncDeclaration search_toString(StructDeclaration sd)
  */
 extern (C++) void semanticTypeInfo(Scope* sc, Type t)
 {
-    extern (C++) final class FullTypeInfoVisitor : Visitor
+    if (sc)
+    {
+        if (!sc.func)
+            return;
+        if (sc.intypeof)
+            return;
+        if (sc.flags & (SCOPE.ctfe | SCOPE.compile))
+            return;
+    }
+
+    if (!t)
+        return;
+
+    void visitVector(TypeVector t)
     {
-        alias visit = Visitor.visit;
-    public:
-        Scope* sc;
+        semanticTypeInfo(sc, t.basetype);
+    }
 
-        override void visit(Type t)
-        {
-            Type tb = t.toBasetype();
-            if (tb != t)
-                tb.accept(this);
-        }
+    void visitAArray(TypeAArray t)
+    {
+        semanticTypeInfo(sc, t.index);
+        semanticTypeInfo(sc, t.next);
+    }
+
+    void visitStruct(TypeStruct t)
+    {
+        //printf("semanticTypeInfo.visit(TypeStruct = %s)\n", t.toChars());
+        StructDeclaration sd = t.sym;
 
-        override void visit(TypeNext t)
+        /* Step 1: create TypeInfoDeclaration
+         */
+        if (!sc) // inline may request TypeInfo.
         {
-            if (t.next)
-                t.next.accept(this);
+            Scope scx;
+            scx._module = sd.getModule();
+            getTypeInfoType(sd.loc, t, &scx);
+            sd.requestTypeInfo = true;
         }
-
-        override void visit(TypeBasic t)
+        else if (!sc.minst)
         {
+            // don't yet have to generate TypeInfo instance if
+            // the typeid(T) expression exists in speculative scope.
         }
-
-        override void visit(TypeVector t)
+        else
         {
-            t.basetype.accept(this);
+            getTypeInfoType(sd.loc, t, sc);
+            sd.requestTypeInfo = true;
+
+            // https://issues.dlang.org/show_bug.cgi?id=15149
+            // if the typeid operand type comes from a
+            // result of auto function, it may be yet speculative.
+            unSpeculative(sc, sd);
```

Comment:
This line was commented before. Is it changed deliberately? During a refactoring?

Reply:
Dang, I just wasted the last hour finding that one! (I don't know how it got uncommented.)

Reply:
(I probably fat-fingered my editor.)